src/main/scala/main.scala

```diff
@@ -54,4 +52,27 @@ object RockSlicer {
     jsonBlocks.saveAsTextFile("blocks.json")
     sc.stop()
   }
+
+  // Produce joints that achieve some lead balancing when generating the initial Block RDD
+  private def generateSeedJoints(joints: Seq[Joint], numSeedJoints: Integer): (Seq[Joint], Seq[Joint]) = {
+    val (persistentJoints, nonPersistentJoints) = joints.partition(_.shape.isEmpty)
+    persistentJoints.length match {
+      // All of the seed joints will be persistent
+      case len if len >= numSeedJoints => (persistentJoints.take(numSeedJoints),
+          persistentJoints.drop(numSeedJoints) ++ nonPersistentJoints)
+      // All persistent joints are used as seed joints, along with some non-persistent joints
+      case len =>
+          // Sort non-persistent joints by relative position along x axis
+          val sortedNonPersistent = nonPersistentJoints.sortWith(_.centerX < _.centerX)
+          val numNonPersistent = numSeedJoints - len
+          val step = sortedNonPersistent.length / numNonPersistent
+          val seedNonPersistentCandidates = (sortedNonPersistent.indices by step) collect sortedNonPersistent
```

Comment:
I think this should be okay, but I might be wrong. Steps in Scala can be tricky. A step isn't the number of elements you skip, it's the difference between included elements. For example:
```scala
scala> (0 to 10 by 3)
res1: scala.collection.immutable.Range = Range(0, 3, 6, 9)
```
So in the case where we only need one non-persistent joint:
```scala
scala> val lst = 0 to 10
lst: scala.collection.immutable.Range.Inclusive = Range(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)

scala> (lst.indices by lst.length) collect lst
res3: scala.collection.immutable.IndexedSeq[Int] = Vector(0)
```

Does this make sense and answer your question?